client/client.go

```diff
@@ -12,7 +12,13 @@ import (
 
 // GetTLSClient creates a Client configured with TLS
 func GetTLSClient(apiURL string, apiKey string, apiVersion string, tlsConfig *tls.Config) (*Client, error) {
-	apiClient, err := getAPIClient(apiURL, apiKey, apiVersion, tlsConfig)
+	var tr http.RoundTripper
+	if tlsConfig != nil {
+		tr = &http.Transport{TLSClientConfig: tlsConfig}
```

Comment:
I know this is not related to your PR, but maybe we can also change it here?
Just initialise `DefaultTransport` and set `TLSClientConfig` afterwards, so it is created with good defaults to catch proxy settings and has timeouts that doesn't last indefinite.

Reply:
Looks like this cannot be done clearly since `DefaultTransport` is an `http.RoundTripper`. I can probably cast it and set the values though.